internal/store/nats.go

```diff
@@ -202,31 +174,23 @@ func (n *natsStore) List(ctx context.Context, srvID uuid.UUID, incState rctypes.
 		return nil, errors.Wrap(ErrRepository, err.Error())
 	}
 
-	var found bool
-	var outgoing rctypes.Condition
-	for _, c := range cr.Conditions {
-		//nolint:gocritic // inverting this test is a moronic suggestion and you should feel bad about it
-		if !rctypes.StateIsComplete(c.State) {
-			// The first incomplete condition is good enough
-			found = true
-			outgoing = *c
-			// make sure to match the requested parameters to short circuit the loops in the handler.
-			// we don't care what the actual state of the found condition is; the fact that it's incomplete
-			// is enough to signal to the API handler that it should *not* accept a new condition
-			// for this server
-			outgoing.Exclusive = true
-			outgoing.State = incState
-			break
-		}
+	if rctypes.StateIsComplete(cr.State) {
+		return nil, nil
 	}
 
-	if !found {
-		return nil, nil
+	var active *rctypes.Condition
+	for _, cond := range cr.Conditions {
+		if !rctypes.StateIsComplete(cond.State) {
+			le.WithFields(logrus.Fields{
+				"condition.id":    cond.ID,
+				"condition.state": string(cond.State),
+			}).Debug("found active condition for server")
+			active = cond
+			break
```

Comment:
Curious we will only have 1 active condition at a time for given server-id right? (Maybe stupid questions just curious and want to learn)

Reply:
Right, as things stand now, having an active condition blocks setting any other condition on a server. It's easier to do this than try and work out what conditions are compatible in a general way.